datashader/core.py

```diff
@@ -192,18 +192,7 @@ def points(self, source, x=None, y=None, agg=None, geometry=None):
         if agg is None:
             agg = count_rdn()
 
-        # Handle down-selecting of SpatialPointsFrame
         if geometry is None:
-            import sys
-            if 'datashader.spatial.points' in sys.modules:
-                from datashader.spatial.points import SpatialPointsFrame
-                if (isinstance(source, SpatialPointsFrame) and
-                        source.spatial is not None and
-                        source.spatial.x == x and source.spatial.y == y and
-                        self.x_range is not None and self.y_range is not None):
-
-                    source = source.spatial_query(
-                        x_range=self.x_range, y_range=self.y_range)
             glyph = Point(x, y)
```

Comment:
This PR seems to remove support for SpatialPointsFrame entirely, while I think we should continue supporting spatial indexing, just no longer assume it's found in Datashader. So I *think* we need to reinstate the deleted code from core.py here, but check for spatialpandas instead. I didn't find SpatialPointsFrame in spatialpandas, though.  @jonmmease , can you suggest what changes we need to make here to avoid depending on the now-deleted spatial module and instead get this functionality from spatialpandas?

Reply:
This logic is specific to the old spatial points frame, so it's not needed when using spatialpandas.

Reply:
Thanks! How does the corresponding functionality work when using SpatialPandas?  I.e., how is the spatial range communicated from Datashader to SpatialPandas in that case?

Reply:
To use the points glyph as an example, the Dask partition-level spatial indexing is done at

https://github.com/holoviz/datashader/blob/dd8f634eb9c61887255f8be80e2354cba55dcef6/datashader/core.py#L209-L215

Here `DaskGeoDataFrame.cx_partitions` is basically equivalent to `SpatialPointsFrame.spatial_query`

The within partition spatial indexing is done in
https://github.com/holoviz/datashader/blob/dd8f634eb9c61887255f8be80e2354cba55dcef6/datashader/glyphs/points.py#L269-L272

Reply:
Perfect, thanks. Looks like this PR is mergeable then!